### DISCUSSION TITLE
Login Oauth Google Failed

### DISCUSSION THREAD

**Discussion by GuilhermeRX** [directus/directus#16386] [Archive]
Hello I'm having trouble configuring oauth login through google on directus, I followed the following tutorial: https://docs.directus.io/self-hosted/sso.html#supported-sso-mechanisms

I set all the environment variables but I still get this error:
<img src='https://i.ibb.co/RvbYwWj/image-1-1.png'/>

.env example

_PUBLIC_URL="my_aplication_host"
AUTH_PROVIDERS="google"
AUTH_GOOGLE_DRIVER="openid"
AUTH_GOOGLE_CLIENT_ID="XXXXX" 
AUTH_GOOGLE_CLIENT_SECRET="XXXX" 
AUTH_GOOGLE_ISSUER_URL="https://accounts.google.com/.well-known/openid-configuration" // try = https://accounts.google.com/
AUTH_GOOGLE_IDENTIFIER_KEY="email"
AUTH_GOOGLE_ICON="google"
AUTH_GOOGLE_LABEL="Google"
AUTH_GOOGLE_ALLOW_PUBLIC_REGISTRATION="true" # 
AUTH_GOOGLE_DEFAULT_ROLE_ID="__role_uuid_"_

#pt-br

**Comment by newathnow** [comment]
Hey fren i had same issue but when i set PUBLIC_URL to the http://localhost:PORT it worked in parts  cause now its return me wrong credentials :D

**Comment by newathnow** [comment]
ok i have forgot the AUTH_GOOGLE_ALLOW_PUBLIC_REGISTRATION="true"  its working just do what i wrote you and will working

**Comment by GuilhermeRX** [comment]
Thank you very much @newathnow it worked perfectly here. ;)
